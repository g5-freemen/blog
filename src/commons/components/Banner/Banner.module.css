.banner {
  width: 100%;
  margin-bottom: 2rem;
  padding: 2rem 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  color: #fff;
  background: var(--cGreen);
  box-shadow: inset 0 8px 8px -8px rgb(0 0 0 / 30%), inset 0 -8px 8px -8px rgb(0 0 0 / 30%);
}

.title {
  font-family: 'Titillium Web';
  font-weight: 700;
  font-size: 3.5rem;
  text-shadow: 0px 1px 3px rgba(0, 0, 0, 0.3);
  line-height: 1;
}

.title:hover {
  animation: pulse infinite 1s alternate ease-in-out;
  cursor: none;
}

@keyframes pulse {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(0.9);
  }
}

.content {
  padding-top: 1rem;
  font-size: 1.5rem;
  font-weight: 300;
}
