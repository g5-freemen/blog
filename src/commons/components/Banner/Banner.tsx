import React from 'react';

import styles from './Banner.module.css';

export default function Banner() {
  return (
    <div className={styles.banner} role="banner">
      <h1 className={styles.title}>conduit</h1>
      <p className={styles.content}>A place to share your knowledge.</p>
    </div>
  );
}
