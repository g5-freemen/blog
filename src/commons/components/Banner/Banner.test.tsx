/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Banner from './Banner';

describe('Banner component', () => {
  beforeEach(() => {
    render(<Banner />);
  });

  test('render h1 conduit', () => {
    const h1tag = screen.getByText(/conduit/i);
    expect(h1tag).toBeInTheDocument();
  });

  test('Banner contains title and text', () => {
    const banner = screen.getByRole('banner');
    expect(banner).toHaveTextContent(/conduit/i);
    expect(banner).toHaveTextContent('A place to share your knowledge');
  });

  test('Banner is match with snapshot', () => {
    const banner = screen.getByRole('banner');
    expect(banner).toMatchSnapshot();
  });
});
