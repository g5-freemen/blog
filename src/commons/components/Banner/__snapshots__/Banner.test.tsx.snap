// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component Banner is match with snapshot 1`] = `
<div
  class="banner"
  role="banner"
>
  <h1
    class="title"
  >
    conduit
  </h1>
  <p
    class="content"
  >
    A place to share your knowledge.
  </p>
</div>
`;
